virttest/utils_iptables.py

```diff
@@ -217,9 +217,16 @@ def command(self, cmd, **dargs):
         self.status, self.output = self.func(self.cmd)
         if self.status != 0:
             logging.error("Failed to execute %s: %s", self.cmd, self.output)
-        # most of the firewall-cmd requires firewalld restart
-        if dargs.get("firewalld_restart", True):
-            self.firewalld_obj.restart()
+
+        # Try to reload the configuration at first instead of
+        # restarting service to avoid of some unexpected errors
+        self.cmd = "%s --reload" % self.firewall_cmd
+        self.status, self.output = self.func(self.cmd)
+        if self.status != 0:
+            logging.error("Failed to execute %s: %s", self.cmd, self.output)
+            # most of the firewall-cmd requires firewalld restart
+            if dargs.get("firewalld_restart", True):
+                self.firewalld_obj.restart()
```

Comment:
@balamuruhans ,
We have to keep 'self.firewalld_obj = Firewalld(session=self.session)', as we need it to initialize the firewalld service state on hosts, for example, when firewalld is not running on remote host. it will help to start the service. Thanks.

Reply:
sure. Thanks @dzhengfy